openml/datasets/dataset.py

```diff
@@ -319,13 +320,15 @@ def _parse_data_from_arff(
         attribute_names = []
         categories_names = {}
         categorical = []
-        for name, type_ in data['attributes']:
+        for i, (name, type_) in enumerate(data['attributes']):
             # if the feature is nominal and the a sparse matrix is
             # requested, the categories need to be numeric
             if (isinstance(type_, list)
                     and self.format.lower() == 'sparse_arff'):
                 try:
-                    np.array(type_, dtype=np.float32)
+                    # checks if the strings which should be the class labels
+                    # can be encoded into integers
+                    pd.factorize(type_)[0]
```

Comment:
Didn't you mention in person that you need to assign the value of this function call?

Reply:
Yes, but with further testing realized that assignment doesn't make sense here since this loop iterates over the _attributes_. Whereas, if anything needs to be checked, we should check the _data_. Which is not seemingly throwing any issue.

I checked this chunk of code with both a _Sparse_Arff_ and _Arff_ data formats, the _type__ receives exactly the same type and structure of the output. I don't know why the attribute list is being checked for type whereas the `arff.ArffDecoder.decode()` seems to return the target feature as a list of the classes. Don't know why a sparse format requires numeric encoding of that attribute list.

Hence, I replaced the numpy check with the pandas categorical encoding.